cli/src/cmd_net.rs

```diff
@@ -0,0 +1,1485 @@
+            "{}\t{}\t{}\t{}\t{}\t{}\t{}\t{}\t{}",
+            "Port".dimmed(),
+            "Configured".dimmed(),
+            "Enabled".dimmed(),
+            "MAC".dimmed(),
+            "Autoneg".dimmed(),
+            "FEC".dimmed(),
+            "Link/FSM State".dimmed(),
+            "Media".dimmed(),
+            "Speed".dimmed(),
+        )?;
+
+        writeln!(
+            &mut mtw,
+            "{}\t{}\t{}",
+            "Receiver Power".dimmed(),
+            "Transmitter Bias Current".dimmed(),
+            "Transmitter Power".dimmed(),
+        )?;
+
+        for p in ports {
+            let status = c
+                .networking_switch_port_status()
+                .port(&p.port_name)
+                .rack_id(p.rack_id)
+                .switch_location(sw)
+                .send()
+                .await
+                .ok()
+                .map(|x| x.into_inner().0);
+
+            let link = status.as_ref().map(|x| {
+                let ls: LinkStatus =
+                    serde_json::from_value(x.get("link").unwrap().clone()).unwrap();
+                ls
+            });
+
+            writeln!(
+                &mut ltw,
+                "{}\t{}\t{}\t{}\t{}\t{}\t{}\t{}\t{}",
+                p.port_name,
+                p.port_settings_id.is_some(),
+                link.as_ref()
+                    .map(|x| x.enabled.to_string())
+                    .unwrap_or("-".to_string()),
+                link.as_ref()
+                    .map(|x| format!(
+                        "{:02x}:{:02x}:{:02x}:{:02x}:{:02x}:{:02x}",
+                        x.address.a[0],
+                        x.address.a[1],
+                        x.address.a[2],
+                        x.address.a[3],
+                        x.address.a[4],
+                        x.address.a[5]
+                    ))
+                    .unwrap_or("-".to_string()),
+                link.as_ref()
+                    .map(|x| x.autoneg.to_string())
+                    .unwrap_or("-".to_string()),
+                link.as_ref()
+                    .map(|x| x.fec.clone())
+                    .unwrap_or("-".to_string()),
+                link.as_ref()
+                    .map(|x| format!("{}/{}", x.link_state, x.fsm_state))
+                    .unwrap_or("-".to_string()),
+                link.as_ref()
+                    .map(|x| x.media.clone())
+                    .unwrap_or("-".to_string()),
+                link.as_ref()
+                    .map(|x| x.speed.clone())
+                    .unwrap_or("-".to_string()),
+            )?;
+
+            let monitors: Option<Monitors> = match status.as_ref() {
+                Some(x) => match x.get("monitors") {
+                    Some(x) => match serde_json::from_value(x.clone()) {
+                        Ok(x) => Some(x),
+                        Err(_) => None,
+                    },
+                    None => None,
+                },
+                None => None,
+            };
```

Comment:
you might check out `than` or `and_then`

Reply:
```suggestion
            let monitors = status
                .as_ref()
                .and_then(|value| value.get("monitors"))
                .and_then(|value| Monitors::deserialize(value).ok());
```

Feel free not to use this, but it uses the sort-of-neat property that a `&serde_json::Value` is a `serde::Deserializer` so whereas `serde_json::from_value` takes a `Value` (no ref), you can just use the ref and not explicitly clone it.

Reply:
Taken